File: src/main/java/com/msg/gauth/domain/email/services/MailVerificationService.java
package com.msg.gauth.domain.email.services;

import com.msg.gauth.domain.email.EmailAuthEntity;
import com.msg.gauth.domain.email.exception.AuthCodeExpiredException;
import com.msg.gauth.domain.email.repository.EmailAuthRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Objects;

@Service
@RequiredArgsConstructor
public class MailVerificationService {
    private final EmailAuthRepository emailAuthRepository;

    @Transactional
    public void execute(String email, String uuid) {
        EmailAuthEntity emailAuth = emailAuthRepository.findById(email)
                .orElseThrow(AuthCodeExpiredException::new);
        if (!Objects.equals(emailAuth.getRandomValue(), uuid))
            throw new AuthCodeExpiredException();
        emailAuth.updateAuthentication(true);
    }
}

File: src/main/kotlin/com/msg/gauth/domain/email/services/MailVerificationService.kt
package com.msg.gauth.domain.email.services

import com.msg.gauth.domain.email.exception.AuthCodeExpiredException
import com.msg.gauth.domain.email.repository.EmailAuthRepository
import com.msg.gauth.global.annotation.service.TransactionalService

@TransactionalService
class MailVerificationService(
    private val emailAuthRepository: EmailAuthRepository
) {
    fun execute(email: String, uuid: String) {
        val emailAuth = emailAuthRepository.findById(email)
            .orElseThrow { AuthCodeExpiredException() }
        if (emailAuth.randomValue != uuid) throw AuthCodeExpiredException()
        val updateEmailAuth = emailAuth.updateAuthentication(true)
        emailAuthRepository.save(updateEmailAuth)
    }
}